src/Nest/CommonAbstractions/SerializationBehavior/SerializerFactory.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Elasticsearch.Net;
+using Newtonsoft.Json;
+
+namespace Nest
+{
+	public interface ISerializerFactory
+	{
+		IElasticsearchSerializer Create(IConnectionSettingsValues settings);
+
+		IElasticsearchSerializer CreateStateful(IConnectionSettingsValues settings, JsonConverter converter);
```

Comment:
I chose to explicitly expose this nasty quality of our serializer here, anyone who implements the factory but does not support the stateful method is not fully adhering to the contract and might get bit (as per the linked multiget issues). To that end i do not think its more solid to split these, these two coexist in a pair.